Tests/Unit/Model/IdealPaymentMethodTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use Wirecard\Oxid\Core\PaymentMethodHelper;
+use Wirecard\Oxid\Model\IdealPaymentMethod;
+use Wirecard\Oxid\Model\SepaCreditTransferPaymentMethod;
+
+use Wirecard\PaymentSdk\Config\Config;
+use Wirecard\PaymentSdk\Config\PaymentMethodConfig;
+use Wirecard\PaymentSdk\Transaction\IdealTransaction;
+
+use OxidEsales\Eshop\Core\Registry;
+
+class IdealPaymentMethodTest extends OxidEsales\TestingLibrary\UnitTestCase
+{
+    /**
+     * @var IdealPaymentMethod
+     */
+    private $_oPaymentMethod;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->_oPaymentMethod = new IdealPaymentMethod();
+    }
+
+    public function testGetConfig()
+    {
+        $oPayment = PaymentMethodHelper::getPaymentById(IdealPaymentMethod::getName(true));
+
+        $oConfig = $this->_oPaymentMethod->getConfig();
+
+        $this->assertInstanceOf(Config::class, $oConfig);
+        $this->assertInstanceOf(PaymentMethodConfig::class, $oConfig->get(IdealPaymentMethod::getName()));
+    }
+
+    public function testGetTransaction()
+    {
+        $aDynArray = [
+            'bank' => 'INGBNL2A',
+        ];
+        Registry::getSession()->setVariable('dynvalue', $aDynArray);
+        $this->assertInstanceOf(IdealTransaction::class, $this->_oPaymentMethod->getTransaction());
+    }
+
+    /**
+     * @dataProvider getNameProvider
```

Comment:
Why does this need a provider if you only test a single data set?

Reply:
Added an additional data to the provider. :+1: